ONSdigital/eq-author-app
# Ear 691 validate non first routing destinations

### What is the context of this PR?

This introduces validation onto routing. It validates:

- If any location which is the destination for a routing rule has been deleted
- If a page or section which is the destination for a routing rule has been moved to an invalid location, e.g. a non-first place within a future section or before the question which has the routing rule on it.

### How to review 

Try the different scenarios and see if the error comes up. Try and break the validation.

* Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
    * ensure it can be opened in Author;
    * then, ensure it can be viewed in Runner by pressing the **view survey** button
* Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Trello card for this task into the next stage of the process
